src/sagemaker/session.py

```diff
@@ -4625,12 +4630,29 @@ def _wait_until_training_done(callable_fn, desc, poll=5):
     return job_desc
 
 
-def _wait_until(callable_fn, poll=5):
-    """Placeholder docstring"""
+def _wait_until(callable_fn, poll_seconds=5, timeout_seconds=None):
+    """
+    Args:
+        callable_fn: callable to wait for which returns None to keep polling
+        poll_seconds (float): time to sleep between calls to callable_fn
+        timeout_seconds (float): Optional stop polling after timeout_seconds elapsed.
+
+    Returns:
+        Result of the callable_fn
+    """
+    waited_seconds = 0.0
+    last_time = time.time()
     result = callable_fn()
-    while result is None:
-        time.sleep(poll)
+    waited_seconds += time.time() - last_time
+    last_time = time.time()
+    while result is None and timeout_seconds and waited_seconds < timeout_seconds:
```

Comment:
Are we sure that the various types of callable_fn that can be passed to this, all would return None?